src/alinea/pyratp/runratp.py

```diff
@@ -73,75 +104,92 @@ def DoAll(*args):
             fichier.write('\n')
-            fichier.write('i_gsca'+"\t")
+            fichier.write('i_gsca' + "\t")
             fichier.write(str(vegetation.i_gsca[jent]))
             fichier.write('\n')
-            fichier.write('agsca'+"\t")
-            fichier.write(str(vegetation.agsca[jent] ))
+            fichier.write('agsca' + "\t")
+            fichier.write(str(vegetation.agsca[jent]))
             fichier.write('\n')
-            fichier.write('i_gslt'+"\t")
+            fichier.write('i_gslt' + "\t")
             fichier.write(str(vegetation.i_gslt[jent]))
             fichier.write('\n')
-            fichier.write('agslt'+"\t")
+            fichier.write('agslt' + "\t")
             fichier.write(str(vegetation.agslt[jent]))
             fichier.write('\n')
-            fichier.write('agsvpd'+"\t")
+            fichier.write('agsvpd' + "\t")
             fichier.write(str(vegetation.agsvpd[jent]))
             fichier.write('\n')
-            fichier.write('avcmaxn'+"\t")
+            fichier.write('avcmaxn' + "\t")
             fichier.write(str(vegetation.avcmaxn[jent]))
             fichier.write('\n')
-            fichier.write('ajmaxn'+"\t")
+            fichier.write('ajmaxn' + "\t")
             fichier.write(str(vegetation.ajmaxn[jent]))
             fichier.write('\n')
-            fichier.write('ardn'+"\t")
+            fichier.write('ardn' + "\t")
             fichier.write(str(vegetation.ardn[jent]))
             fichier.write('\n')
-            fichier.write('ismine'+"\t")
+            fichier.write('ismine' + "\t")
             fichier.write(str(vegetation.ismine[jent]))
             fichier.write('\n')
-            fichier.write('epm'+"\t")
+            fichier.write('epm' + "\t")
             fichier.write(str(vegetation.epm[jent]))
             fichier.write('\n')
 
-
         fichier.close()
 
-        return ratp.out_time_spatial, ratp.out_time_tree
+        # convert matrices to dataframes
+        df = pd.DataFrame(ratp.out_time_spatial, columns=columns)
+
+        df['vtyp'] = [int(t) for t in df['VegetationType']]
+        df['ntime'] = [int(t) for t in df['ntime']]
+        df['day'] = [int(t) for t in df['day']]
+        df['vid'] = [int(t) - 1 for t in df['VoxelId']]
+
+        df_spatial = df.set_index(['vtyp', 'ntime', 'vid'])
+
+        df = pd.DataFrame(ratp.out_time_tree, columns=columns_tree)
+
+        df['vtyp'] = [int(t) for t in df['VegetationType']]
+        df['ntime'] = [int(t) for t in df['ntime']]
+        df['day'] = [int(t) for t in df['day']]
+
+        df_tree = df.set_index(['vtyp', 'ntime'])
+
+        return df_spatial, df_tree
 
     @staticmethod
     def DoIrradiation(*args):
         ratp = pyratp.ratp
-        ratp.out_rayt = np.zeros(pyratp.micrometeo.nbli*pyratp.grid3d.nveg*9*pyratp.grid3d.nent).reshape(pyratp.micrometeo.nbli*pyratp.grid3d.nveg*pyratp.grid3d.nent ,9)
+        ratp.out_rayt = np.zeros((pyratp.micrometeo.nbli * pyratp.grid3d.nveg * pyratp.grid3d.nent, 9))

```

Comment:
Here, what 9 means?

Reply:
no idea, I just took the opportunity of this branch to change a reshape by directly passing the shape to np.zeros :)
But I'll investigate